src/libraries/System.Net.Security/tests/FunctionalTests/TestConfiguration.cs

```diff
@@ -16,7 +16,7 @@ namespace System.Net.Security.Tests
     internal static class TestConfiguration
     {
         public const int PassingTestTimeoutMilliseconds = 4 * 60 * 1000;
-        public const int FailingTestTimeoutMiliseconds = 500;
+        public const int FailingTestTimeoutMiliseconds = 5000;
```

Comment:
How much time does this add to execution?  As I've mentioned previously, generally we've used the "Failing" test timeout for tests that need to wait for this full amount of time to assert that something hasn't happened, whereas "Passing" test timeout is used to assert that something has happened... thus we generally have a long passing timeout (because we want to give the test a lot of time for the thing to happen) and a very short failing timeout (because we can't wait forever for something that's not supposed to happen to happen).  It thus concerns me that we're increasing a failing test timeout to 5s.  That suggests to me we're either using it incorrectly or something more fundamental is happening that should be addressed.

Reply:
no extra time as far as I can tell. The 12 variations of this test took 0.233s on my VM. (see the description) 
I'm wondering if the naming is confusing. The test verifies that something happened e.g. in this case we sent alert and exception is raised.  It is labeled failing because handshake did not completed but it fits your description of "Passing".

From what I can tell, the time variation comes from networking and IO. Alternatively, we can switch to memory stream or maybe UDS.  With the helper we could be more flexible. The current TCP is nice for investigation as Wireshark can decode everything as well as it is closer to how people use it in production.

Reply:
> I'm wondering if the naming is confusing. The test verifies that something happened e.g. in this case we sent alert and exception is raised. It is labeled failing because handshake did not completed but it fits your description of "Passing".

Based on your description, it should be using "Passing".  The test expects something to happen and the test will fail if it doesn't; that's what the "Passing" timeout is for.  When a test expects something to _not_ happen and needs to wait to validate that it doesn't, that's what the "Failing" timeout is for: we can't wait forever, so we pick a reasonable time after which we say "if it didn't happen yet, we'll just assume it won't", and we don't want it to be too long, because the longer it is, the longer we'll sit waiting.

Feel free to rename if you have better names for them.  But we should switch to use the appropriate timeouts rather than increasing Failing.

Reply:
updated.